hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileInfo.java

```diff
@@ -306,16 +344,13 @@ ReaderContext createReaderContext(boolean doDropBehind, long readahead, ReaderTy
         // Intercept the exception so can insert more info about the Reference; otherwise
         // exception just complains about some random file -- operator doesn't realize it
         // other end of a Reference
-        FileNotFoundException newFnfe = new FileNotFoundException(toString());
-        newFnfe.initCause(fnfe);
-        throw newFnfe;
+        throw decorateFileNotFoundException(fnfe);
       }
-      status = fs.getFileStatus(referencePath);
     } else {
       in = new FSDataInputStreamWrapper(fs, this.getPath(), doDropBehind, readahead);
-      status = fs.getFileStatus(initialPath);
     }
-    long length = status.getLen();
+    getFileStatus();
```

Comment:
remove this line since it looks redundant?

Reply:
+1